manifests/sysusers.pp

```diff
@@ -0,0 +1,27 @@
+# @summary  Prepare systemd sysusers files
+#
+# @api public
+#
+# @param purgedir If true /etc/sysusers.d will be purged of unmanaged files.
+#
+# @see systemd-sysusers(8)
+#
+class systemd::sysusers (
+  Boolean $purgedir = true,
+) {
+  file { '/etc/sysusers.d':
+    ensure  => directory,
+    owner   => root,
+    group   => root,
+    mode    => '0755',
+    purge   => $purgedir,
+    recurse => $purgedir,
+    force   => $purgedir,
+  }
+
+  exec { 'systemd-sysusers':
+    command     => 'systemd-sysusers',
+    refreshonly => true,
+    path        => $facts['path'],
+  }
```

Comment:
Should the `systemd-sysusers` exec run if the file resource removes some files from `/etc/sysusers.d`?

Reply:
It's harmless but indeed pointless. I will remove the notify for that case.

Reply:
Oh, I did not see a notify for that case, I was wondering if there needs to be one 🙂